File: 3467-transform-array-by-parity/3467-transform-array-by-parity.java
class Solution {
    public int[] transformArray(int[] nums) {  
        int[] arr=new int[nums.length];
        Arrays.fill(arr,1);  
        int a=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]%2==0){
                arr[a++]=0;
            }
        }
        return arr;
    }
}

File: 3467-transform-array-by-parity/3467-transform-array-by-parity.py
class Solution:
    def transformArray(self, nums: List[int]) -> List[int]:
        n=len(nums)
        arr=[1]*n
        a=0
        for i in range(n):
            if nums[i]%2==0:
                arr[a]=0
                a+=1
        return arr